drivers/input/djoystick.c

```diff
@@ -524,12 +459,7 @@ static ssize_t djoy_read(FAR struct file *filep, FAR char *buffer,
 
   /* Get exclusive access to the driver structure */
 
-  ret = djoy_takesem(&priv->du_exclsem);
-  if (ret < 0)
-    {
-      ierr("ERROR: djoy_takesem failed: %d\n", ret);
-      return ret;
-    }
+  flags = enter_critical_section();
```

Comment:
Is `dl_sample` implementation ready that now it is called from inside the critical section?

Reply:
Yes, since dl_sample is already be called inside the critical secition in djoy_sample:
https://github.com/apache/incubator-nuttx/pull/5204/files#diff-cb78728fc9ab1e7ff74388d2382edcc0701a691fa0d737dcffce197a2eb8f8baR249-R254